src/main/daml/Daml/Finance/Instrument/Bond/FixedRate/Factory.daml

```diff
@@ -64,4 +60,4 @@ template Factory
       view = Disclosure.View with disclosureControllers = singleton provider; observers
       setObservers Disclosure.SetObservers{newObservers} =
         toInterfaceContractId <$> create this with observers = newObservers
-      archive' self = archive (coerceContractId self : ContractId FixedRate.Instrument)
+      archive' = archive . fromInterfaceContractId @Factory
```

Comment:
I see a patten of `archive'` implementations as part of the `Disclosure` interface on factories that they coerce to the underlying instrument (in this case `FixedRate.Instrument`) instead of the `Factory`. My understanding is that this would throw a runtime exception as a factory cannot be coerced to the underlying instrument.

Reply:
🤔 good catch!

Reply:
Unfortunately we don't have tests for that currently